src/utils/get-base-and-head-commit-shas.ts

```diff
@@ -14,13 +14,18 @@ export const getBaseAndHeadCommitShas = async (
   if (event.type === "pull_request") {
     const head = event.payload.pull_request.head.sha;
     const base = event.payload.pull_request.base.sha;
+    const baseRef = event.payload.pull_request.base.ref;
     if (options.useDeploymentUrl) {
       // Vercel deploys the head commit of the PR, not the github temporary merge commit
-      // So the PR base is the correct base commit to use
-      return { base, head };
+      // The PR base can sometimes point to a commit ahead of the merge-base of the head commit
+      // (I believe it's based on the github temporary merge commit)
```

Comment:
The PR base seems to be neither guaranteed be the base commit of the temporary merge commit nor the base commit of the head commit. I wonder if it's intended to be the base commit of the temporary merge commit but is just sometimes out of date? e.g. github doesn't instantly update it after creating a new temporary merge commit

Reply:
ChatGPT's answer is plausible:

> When using the github API, the PR base ("pull_request.base.sha") seems to be neither guaranteed be the merge-base of the temporary merge commit ("GITHUB_SHA") nor the merge-base of the head commit ("pull_request.head.sha"). How is "pull_request.base.sha" calculated?
> pull_request.base.sha is the latest commit on the base branch (i.e., the branch you are proposing to merge into) at the time the pull request is created. It is not guaranteed to be the merge-base of the temporary merge commit (GITHUB_SHA) or the merge-base of the head commit (pull_request.head.sha).
> 
> Here's a brief explanation of each:
> 
> pull_request.base.sha: The latest commit on the base branch at the time the pull request is created. It may change if the base branch is updated after the pull request is created.
> 
> GITHUB_SHA: The SHA of the temporary merge commit created by GitHub Actions when checking out the PR. This commit represents the result of merging the head branch into the base branch at the time the action is run. It may change if the base branch or the head branch are updated.
> 
> pull_request.head.sha: The latest commit on the head branch (i.e., the branch with the changes you want to merge) at the time the pull request is created. It may change if new commits are pushed to the head branch after the pull request is created.
> 
> Since the base branch and head branch can both be updated after the pull request is created, it's possible for the pull_request.base.sha to not match the merge-base of either the temporary merge commit or the head commit.

I haven't found a good specification of it in the Github docs.

Reply:
Note that I use merge-base while https://github.com/JensDll/should-run/blame/main/.github/workflows/main.yaml#L54 uses rev-list (return the parent of the oldest commit that is reachable from pull_request.head.sha but not from pull_request_base.sha). But I think both methods should work with both merge commits and re-bases.